content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,595 @@
+        assertEquals(3, cart.totalNumberOfItems());
+        cart.removeProductsFromCartByCustomer(1L, product);
+        assertEquals(2, cart.totalNumberOfItems());
+
+
+    }
+
+    @Test
+    void calculatingAllTheProductsInTheCart() {
+        Product product = new Product("1", "Sandine", new BigDecimal(1000);
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Customer customer=new Customer(1L, "Kingsley", "Chukwudi", new BigDecimal("2000"));
+        dataService.registerCustomer(customer);
+        cart.addProductToCartByCustomer(1L, product);
+        cart.addProductToCartByCustomer(1L, product2);
+        cart.addProductToCartByCustomer(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+
+        assertEquals(new BigDecimal( 3100), cart.calculatePrice());
+    }
+    @Test
+    void addProductToCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+    }
+    @Test
+    void removeProductsFromCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        System.out.println(cart);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart.removeProductsFromCartByAdmin(1L,product);
+        System.out.println(cart);
+        assertEquals(2,cart.totalNumberOfItems());
+
+
+    }
+    @Test
+    void checkAllTheProductsInTheCartByCustomer(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Product product3 = new Product("3", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product3);
+        System.out.println(  cart.checkAllTheProductsInTheCartByCustomer(1L));
+    }
+    @Test
+    void checkAllRegisteredAdmin(){
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        System.out.println(dataService.getAllAdmin());
+    }
+}
```

Comment:
You forgot to implement this feedback I gave [here](https://github.com/section-engineering-education/engineering-education/pull/6859#discussion_r817708483)

Reply:
@john-amiscaray I actually did a field dependency injection may be due to copy and pastes I might have omitted it.

Reply:
No, I was talking about this comment I left:

> This section on integration testing assumes that the user knows how to write JUnit tests in the context of Spring. This may not be the case so I would either add some explanation about how the testing works (i.e., the assertions, the @SpringBootTest annotation, and the @Test annotation, or I would add some short comments in the code talking about them.